homeassistant/components/nut/__init__.py

```diff
@@ -148,7 +148,7 @@ def _serial_from_status(status: dict[str, str]) -> str | None:
     """Find the best serialvalue from the status."""
     serial = status.get("device.serial") or status.get("ups.serial")
     if serial and (
-        serial.lower() in NUT_FAKE_SERIAL or serial.count("0") == len(serial)
+        serial.lower() in NUT_FAKE_SERIAL or serial.count("0") == len(serial.strip())
```

Comment:
Why isn't the serial strip:ed directly before any check is done. We don't want trailing zero's propagating into unique_ids later anyway.

Reply:
Ok, make sense, but for this I will also change a little bit the function `_unique_id_from_status` that compose the unique_id.

Reply:
But this could be also considered a breaking change, because could change the unique_id generated for already configured devices. Are you sure that I should proceed?

Reply:
No not really. Would need an upgrade step. So lets do that separate.